sdk/bootstrap/main.go

```diff
@@ -132,47 +197,62 @@ func waitForResolve(resolveHosts []string, resolveTimeout time.Duration) {
 	} else {
 		timer = time.NewTimer(resolveTimeout)
 	}
+
 	for _, host := range resolveHosts {
-		log.Printf("Waiting for '%s' to resolve...", host)
-		for {
-			result, err := net.LookupHost(host)
-
-			// Check result, exit loop if suceeded:
-			if err != nil {
-				if verbose {
-					log.Printf("Lookup failed: %s", err)
-				}
-			} else if len(result) == 0 {
-				if verbose {
-					log.Printf("No results for host '%s'", host)
-				}
-			} else {
-				log.Printf("Resolved '%s' => %s", host, result)
-				break
-			}
+		resolveHost(host, timer)
+	}
+
+	if verbose {
+		log.Printf("Hosts resolved, continuing bootstrap.")
+	}
+
+	// Clean up:
+	if !timer.Stop() {
+		<-timer.C
+	}
+}
+
+func verifySelfResolution(podIP string, resolveTimeout time.Duration) {
+	taskHost, err := parseTaskHost()
+	if err != nil {
+		printEnv()
+		log.Printf("Missing required envvars to build task DNS address. " +
```

Comment:
Is the intention to continue execution here? Should we exit or break?

Reply:
Whoops. That should be a Fatalf.